runtime/compiler/control/CompilationThread.cpp

```diff
@@ -7414,6 +7414,11 @@ TR::CompilationInfoPerThreadBase::wrappedCompile(J9PortLibrary *portLib, void *
                   options->setOption(TR_EnableGRACostBenefitModel, false);
                }
 
+            if (jitConfig->javaVM->phase != J9VM_PHASE_NOT_STARTUP || options->getOptLevel() < warm)
+               {
```

Comment:
Assuming the symbolValidationManager is the default, maybe turning it off for AOT loads does not make sense. Let the AOT load do whatever it needs to do based on how the code was generated. This will eliminate the need to set this option later in prepareRelocateAOTCodeAndData

Reply:
This code is guarded by `if (!that->_methodBeingCompiled->_aotCodeToBeRelocated) // exclude AOT loads` so it never runs under AOT loads. In fact, on an AOT load, regardless of whether we want to use the new infra or not, if the body we're relocating has validations from the new infra, we have to validate it using the new infra (hence the addition of the flag in the AOT Method Header).